src/modules/auth/dto/contract.dto.ts

```diff
@@ -0,0 +1,46 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsArray, IsNotEmpty, IsString, IsUUID } from 'class-validator';
+
+export class CallContractInputDto {
+  @IsNotEmpty()
+  @IsUUID()
+  token: string;
+
+  @ApiProperty({
+    example: 'myChannel',
+    description: 'Name of the channel that has the contract',
+    required: true,
+  })
+  @IsNotEmpty()
+  @IsString()
+  channelName: string;
+
+  @ApiProperty({
+    example: 'paymentContract',
+    description: 'Name of the contract containing the function',
+    required: true,
+  })
+  @IsNotEmpty()
+  @IsString()
```

Comment:
Since we have a lot of such `RequiredFields` in our schemas, let's create a Decorator for `RequiredTypes` that will have these 3 properties. 

Duplicating them on all fields makes the code very clustered.

Reply:
sure